assetconnection/opcua/src/main/java/de/fraunhofer/iosb/ilt/faaast/service/assetconnection/opcua/provider/OpcUaOperationProvider.java

```diff
@@ -130,18 +142,36 @@ private Argument[] getOutputArguments(UaMethodNode node) throws AssetConnectionE
                     : new Argument[0];
         }
         catch (InterruptedException | ExecutionException e) {
-            Thread.currentThread().interrupt();
-            throw new AssetConnectionException(String.format("Could not read output arguments (nodeId: %s)",
-                    providerConfig.getNodeId()),
-                    e);
+            if ((e.getCause() instanceof UaException) && (((UaException) e.getCause()).getStatusCode().getValue() == StatusCodes.Bad_NotFound)) {
+                return new Argument[0];
+            }
+            else {
+                Thread.currentThread().interrupt();
+                throw new AssetConnectionException(String.format("Could not read output arguments (nodeId: %s)",
+                        providerConfig.getNodeId()),
+                        e);
+            }
         }
     }
 
 
     private void init() throws AssetConnectionException {
         nodeId = OpcUaHelper.parseNodeId(client, providerConfig.getNodeId());
         final UaMethodNode methodNode = getMethodNode(nodeId);
-        parentNodeId = getParentNode(nodeId).getNodeId();
+        if (providerConfig.getParentNodeId() != null) {
```

Comment:
I took the suggestion, even if it is not equivalent to my code. But it should work in this case. 
A question in this context would be, if we accept an empty string for the parentNodeId? 
Currently, the default parent is used, when the given parentNodeId is null. An empty string for the parentNodeId throws an exception, because it's not a valid NodeId.

Reply:
IMHO we should treat an empty string for a parentNodeId like a null value

Reply:
That's also my opinion. 
Now an empty string for the parentNodeId is treated like a null pointer